JASP-Engine/JASP/R/classicalmetaanalysis.R

```diff
@@ -124,39 +124,49 @@ ClassicalMetaAnalysis <- function(dataset=NULL, options, perform="run", callback
     )
     
     .hasErrors(dataset, perform, type=c("infinity", "observations"),
-      all.target=numeric.variables, observations.amount="< 2",
-      exitAnalysisIfErrors=TRUE)
+               all.target=numeric.variables, observations.amount="< 2",
+               exitAnalysisIfErrors=TRUE)
   }
   
   method = switch(options$method, `Fixed Effects` = "FE", `Maximum Likelihood` = "ML",
-      `Restricted ML` = "REML", `DerSimonian-Laird` = "DL", `Hedges` = "HE",
-      `Hunter-Schmidt` = "HS", `Sidik-Jonkman` = "SJ", `Empirical Bayes` = "EB",
-      `Paule-Mandel` = "PM", "REML")
-
+                  `Restricted ML` = "REML", `DerSimonian-Laird` = "DL", `Hedges` = "HE",
+                  `Hunter-Schmidt` = "HS", `Sidik-Jonkman` = "SJ", `Empirical Bayes` = "EB",
+                  `Paule-Mandel` = "PM", "REML")
+  
   can.run = all(c(effsizeName, stderrName) != "")
   
   ## Run the analysis. ####
   if (is.null(rma.fit)) {
-    rma.fit <- structure(list('b' = numeric(),'se' = numeric(),'ci.lb' = numeric(),'ci.ub' = numeric(), 
+    rma.fit <- structure(list('b' = numeric(),'se' = numeric(),'ci.lb' = numeric(), 'ci.ub' = numeric(), 
                     'zval' = numeric(),'pval' = numeric()), class = c("dummy", "rma"))
     
     if (run && can.run) {
       
+      # infer model formula
       .vmodelTerms = options$modelTerms
       formula.rhs <- as.formula(as.modelTerms(.vmodelTerms))
       if (is.null(formula.rhs)) 
         formula.rhs = ~1
+      
       if (!options$includeConstant) 
         formula.rhs = update(formula.rhs, ~ . + 0)
-      if( identical(formula.rhs, ~ 1 - 1) )
+      
+      if (identical(formula.rhs, ~ 1 - 1))
         .quitAnalysis("The model should contain at least one predictor or an intercept.")
-
-      rma.fit <- metafor::rma(
+      
+      # check input data
+      if (any(dataset[[b64(stderrName)]] <= 0))
+        .quitAnalysis(paste0("Non-positive standard errors encountered in variable '", stderrName, "'. Standard errors should be positive. Please check your input."))
+      
+      # analysis
+      rma.fit <- tryCatch( # rma generates informative error messages; use them!
+        metafor::rma(
           yi = get(b64(effsizeName)), sei = get(b64(stderrName)), data = dataset,
           method=method, mods = b64(formula.rhs), test = options$test,
-          slab = if(options$studyLabel != "") get(b64(options$studyLabels)),
-          level = options$regressionCoefficientsConfidenceIntervalsInterval
-        )
+          slab = if(options$studyLabel != "") paste0(get(b64(options$studyLabels))),
+          level = options$regressionCoefficientsConfidenceIntervalsInterval + 1e-9, # add tiny amount because 1 is treated by rma() as 100% whereas values > 1 as percentages
+          control = list(maxiter = 500)
+        ), error = .quitAnalysis)
```

Comment:
Still have some doubts about the blanket tryCatch(), e.g.:

`Error in metafor::rma(yi = get(b64(effsizeName)), sei = get(b64(stderrName)), : Fisher scoring algorithm did not converge. See 'help(rma)' for possible remedies.`
Gives a lot of useless information. Now if we change the error condition a little:
```
), error = function(e) .quitAnalysis(e$message))
```
We obtain a slightly better message:
`Fisher scoring algorithm did not converge. See 'help(rma)' for possible remedies.`
Although the 'help(rma)' bit is still a bit evil, because this does not help JASP users.

We'll keep it because in general the messages are quite good, but please add the code suggestion.